src/control/sessions.rs

```diff
@@ -0,0 +1,34 @@
+use std::fmt::Debug;
+
+use axum::{Extension, Json, http::StatusCode};
+use axum_extra::extract::{CookieJar, cookie::Cookie};
+use cookie::time::OffsetDateTime;
+use tracing::info;
+
+use crate::{config::CONFIG, domain::users::Credentials, service::sessions::{SessionService, LoginError}};
+
+#[tracing::instrument(skip_all, fields(email = credentials.email))]
+pub async fn post_sessions<T: SessionService + Debug>(
+    Extension(service): Extension<T>,
+    jar: CookieJar,
+    Json(credentials): Json<Credentials>
+) -> Result<(CookieJar, StatusCode), StatusCode> {
+    info!("Received login attempt");
+    let session = match service.login(credentials).await {
+        Err(LoginError::NoUser) =>  {
+            return Err(StatusCode::UNAUTHORIZED);
+        },
+        Err(LoginError::Unknown) => {
+            return Err(StatusCode::INTERNAL_SERVER_ERROR);
+        },
+        Ok(session) => session
+    };
+
+    let cookie = Cookie::build(&CONFIG.session_cookie_name, session.id)
+        .expires(OffsetDateTime::from_unix_timestamp(session.expires).unwrap())
+        .http_only(true)
+        // .secure(true) <-- add this when TLS is set up
```

Comment:
Does it require that all traffic uses https? Right now only app <-> gateway communication will be encrypted, but all other will use standard http

Reply:
Makes sense, this doesn't have to be encrypted

Reply:
Actually, this is the header that says what cookie should be set by the end user, so yeah it should require `Secure`.

Reply:
Ok, to be continued when we will enable tls